bases/renku_data_services/data_api/posthog.py

```diff
@@ -1,120 +1,57 @@
-        if not self.enabled:
-            return
-        self._posthog.capture(distinct_id=self._anonymize_user_id(user), event="data_connector_created")
-
-    async def data_connector_linked(self, user: APIUser) -> None:
-        """Send data connector linked event to metrics."""
-        if not self.enabled:
-            return
-        self._posthog.capture(distinct_id=self._anonymize_user_id(user), event="data_connector_linked")
-
-    async def project_member_added(self, user: APIUser) -> None:
-        """Send project member added event to metrics."""
-        if not self.enabled:
-            return
-        self._posthog.capture(distinct_id=self._anonymize_user_id(user), event="project_member_added")
-
-    async def group_created(self, user: APIUser) -> None:
-        """Send group created event to metrics."""
-        if not self.enabled:
-            return
-        self._posthog.capture(distinct_id=self._anonymize_user_id(user), event="group_created")
-
-    async def group_member_added(self, user: APIUser) -> None:
-        """Send group member added event to metrics."""
-        if not self.enabled:
-            return
-        self._posthog.capture(distinct_id=self._anonymize_user_id(user), event="group_member_added")
-
-    async def search_queried(self, user: APIUser) -> None:
-        """Send search queried event to metrics."""
-        if not self.enabled:
-            return
-        self._posthog.capture(
-            distinct_id=self._anonymize_user_id(user),
-            event="search_queried",
-            properties={"authenticated": user.is_authenticated},
-        )
+"""Sanic task for processing metrics data and sending to PostHog."""
+
+import asyncio
+
+import uvloop
+from sanic.log import logger
+
+from renku_data_services.app_config import Config
+
+
+async def _send_metrics_to_posthog() -> None:
+    from posthog import Posthog
+
+    config = Config.from_env()
+
+    posthog = Posthog(
+        api_key=config.posthog.api_key,
+        host=config.posthog.host,
+        sync_mode=True,
+        super_properties={"environment": config.posthog.environment},
+    )
+
+    while True:
+        try:
+            metrics = config.metrics_repo.get_unprocessed_metrics()
+
+            processed_ids = []
+            async for metric in metrics:
+                try:
+                    posthog.capture(
+                        distinct_id=metric.anonymous_user_id,
+                        timestamp=metric.timestamp,
+                        event=metric.event,
+                        properties=metric.metadata_ or {},
+                        # This is sent to avoid duplicate events if multiple instances of data service are running.
+                        # Posthog deduplicates events with the same timestamp, distinct_id, event, and uuid fields:
+                        # https://github.com/PostHog/posthog/issues/17211#issuecomment-1723136534
+                        uuid=metric.id,
```

Comment:
```suggestion
                        uuid=metric.id.to_uuid4(),
```
if it's not a valid uuid (which ULID is not), posthog rejects the request.